### PR TITLE
Stamp date on subsequent error modals.

### PR DESCRIPTION
**Pull request by galegosimpatico**
Before this change modals stack on the same exact place, so the user
does not receive visual feedback until all of them are closed, in case
the errored action is requested over and over again.

This change adds a stamp to subsequent error modals, so the user would
receive visual feedback every time one of the error modals is closed.

**At the time of writing this code hasn't been run yet.**

### CODE REVIEW

File: app/assets/javascripts/qpixel_api.js [hunk 1]
```diff
@@ -12,15 +12,42 @@ window.QPixel = {
     return token;
   },
 
+  /** Counts notifications popped up at any time. */
+  var popped_modals_ct = 0;
```
Comment (ArtOfCode-) [inline app/assets/javascripts/qpixel_api.js:16]: We're using ES6, so `const` and `let` - `const` if the value won't change, `let` if you need to re-assign to the variable later.

File: app/assets/javascripts/qpixel_api.js [hunk 2]
```diff
@@ -12,15 +12,42 @@ window.QPixel = {
...
+
   /**
    * Create a notification popup - not an inbox notification.
    * @param type the type to apply to the popup - warning, danger, etc.
    * @param message the message to show
    */
   createNotification: function(type, message) {
+    /** Some messages include a date stamp. */
+    var append_date = false;
```
Comment (ArtOfCode-) [inline app/assets/javascripts/qpixel_api.js:25]: Likewise `const` and `let` - just assume I've repeated this for every instance of `var` :)

File: app/assets/javascripts/qpixel_api.js [hunk 3]
```diff
@@ -12,15 +12,42 @@ window.QPixel = {
...
+  /** Counts notifications popped up at any time. */
+  var popped_modals_ct = 0;
+
   /**
    * Create a notification popup - not an inbox notification.
    * @param type the type to apply to the popup - warning, danger, etc.
    * @param message the message to show
    */
   createNotification: function(type, message) {
+    /** Some messages include a date stamp. */
```
Comment (ArtOfCode-) [inline app/assets/javascripts/qpixel_api.js:24]: Single-line comments are `//`

File: app/assets/javascripts/qpixel_api.js [hunk 4]
```diff
@@ -12,15 +12,42 @@ window.QPixel = {
...
+    var message_with_date = message;
+    var span, button;
+    if (type === 'danger') {
+      if (popped_modals_ct > 0 ) {
+        /* Modals stack each on top, so repeating an errored action over
+         * and over again is gonna create multiple error modals in the
+         * same exact place. While this happens this way, an user
+         * closing the error modal will not have an immediate visual
+         * action feedback if two or more error modals are printed.
+         * A date is stamped in order to cope with that. */
```
Comment (ArtOfCode-) [inline app/assets/javascripts/qpixel_api.js:35]: Max characters per line is 120, including comments, so these lines can be longer and the comment won't span so many lines.

**Review by ArtOfCode- — CHANGES_REQUESTED** [review]

**Comment by galegosimpatico** [comment]
remains unrun from my side for now, but if everything goes well with my local setup, now I should be able to check

**Comment by galegosimpatico** [comment]
I've seen the tags cache (what I think it is) being populated before, but for some reason not today, not even when creating a brand new category and tags set, clearing cache, restarting server... 🤔

**Comment by galegosimpatico** [comment]
Now I have seen it working.

File: app/assets/javascripts/qpixel_api.js [hunk 5]
```diff
@@ -12,15 +12,40 @@ window.QPixel = {
...
+    // Some messages include a date stamp, `append_date` governs that.
+    let append_date = false;
+    let message_with_date = message;
+    if (type === 'danger') {
+      if (popped_modals_ct > 0 ) {
+        /* At the time of writing, modals stack each one exactly on top of previous one, so repeating an errored action
+         * over and over again is going to create multiple error modals in the same exact place. While this happens this
+         * way, an user closing the error modal will not have an immediate visual action feedback if two or more error
+         * modals have been printed. A date is stamped in order to cope with that. Could be anything. Probablye a cycle
+         * of different emoji characters would be cuter while having the purpose met. But if so make sure character in
```
Comment (galegosimpatico) [inline app/assets/javascripts/qpixel_api.js:33]: obviously insperd by discord's this thing

Screenshot 2020-08-26 at 02 12 24

Screenshot 2020-08-26 at 02 12 30

File: app/assets/javascripts/qpixel_api.js [hunk 6]
```diff
@@ -12,9 +15,6 @@ window.QPixel = {
     return token;
   },
 
-  /** Counts notifications popped up at any time. */
-  let popped_modals_ct = 0;
```
Comment (galegosimpatico) [inline app/assets/javascripts/qpixel_api.js]: https://thecodinglove.com/when-i-run-my-code-for-the-first-time

File: app/assets/javascripts/qpixel_api.js (hunk 2 above)
Comment (galegosimpatico) [reply app/assets/javascripts/qpixel_api.js]: `var` in https://github.com/codidact/landing-page/pull/53/files too?

Comment (ArtOfCode-) [reply app/assets/javascripts/qpixel_api.js]: I have yet to review that one properly, but probably :)

**Pull request merged**